static/app/views/insights/sessions/components/insightLayoutContext.tsx

```diff
@@ -0,0 +1,68 @@
+import type {ReactNode} from 'react';
+import {createContext, useCallback, useContext, useEffect, useState} from 'react';
+
+import localStorage from 'sentry/utils/localStorage';
+import type {DomainView} from 'sentry/views/insights/pages/useFilters';
+import {
+  type CHART_MAP,
+  DEFAULT_LAYOUTS,
+  PAGE_CHART_OPTIONS,
+} from 'sentry/views/insights/sessions/components/settings';
+
+type TChart = keyof typeof CHART_MAP;
+type TCharts = readonly TChart[];
```

Comment:
Also wondering if maybe this should be a tuple, since the number of charts is known? e.g., `[TChart, TChart, TChart, TChart, TChart]` instead of `TChart[]` so it's not possible to set an incorrect number of chart selections